orttraining/orttraining/python/ort_trainer.py

```diff
@@ -326,11 +326,26 @@ def convert_model_loss_fn_to_onnx(model, loss_fn, model_desc, device, inputs, op
                        do_constant_folding=False,
                        **other_export_options)
 
-    model = onnx.load_model_from_string(f.getvalue())
-
-    model = FuseSofmaxNLLToSoftmaxCE(model)
+    onnx_model = onnx.load_model_from_string(f.getvalue())
+
+    # Remove 'model_.' prefix introduced by model wrapper for initializers.
+    replace_name_dict = {}
+    for n in onnx_model.graph.initializer:
+        if n.name.startswith('model_.'):
+            replace_name_dict[n.name] = n.name[len('model_.'):]
+            n.name = replace_name_dict[n.name]
+    for n in onnx_model.graph.node:
+        for i, name in enumerate(n.input):
+            if name in replace_name_dict:
+                n.input[i] = replace_name_dict[name]
+    assert set([n.name for n in onnx_model.graph.initializer]) == \
+        set([n for n, t in model.model_.named_parameters()]), \
```

Comment:
One problem with this assert that I found while testing is that torch named_parameters()  contain just the trainable named_parameters(), while the onnx model initializers also contains the non-trainable torch 'buffers'.

Reply:
good catch, thanks